APP-7464 support unix sockets on windows OS

abe-winter opened the pull request:
## What changed
- add cleanWindowsUDS function to clean unix socket paths on windows hosts, so they're compatible w/ the grpc library
- still respect VIAM_TCP_SOCKETS env var, but don't force-activate it on goos=windows
## Why
The `url.Parse` call in the (3rd-party) grpc library, [here](https://github.com/grpc/grpc-go/blob/v1.71.0/clientconn.go#L1720-L1727), as well as its direct caller, become unhappy in various ways when given windows-looking filesystem paths. This PR works around that by formatting the paths in a way it likes.

To see this fail in the small, do:

```golang
urls := []string{
	"unix://C:\\hello\\whatever.sock",
	"unix://\\hello\\whatever.sock",
	"unix://C:/hello/whatever.sock",
	"unix:///hello/whatever.sock",
}
for i, raw := range urls {
	if u, err := url.Parse(raw); err != nil {
		println("err", i, err.Error())
	} else {
		println("result", i, u.String())
	}
}
```

(The first two fail, the last two succeed).

I'm stripping the `C:` prefix as well because I get some kind of 'invalid authority' error from grpc when I pass that.
## Notes
### Future work
The `VIAM_TCP_SOCKETS` flag is still respected when present. I think we should remove this feature completely.
### Compatibility with existing windows modules
This seems to work with the windows modules that are currently on the registry and which we've tested in TCP mode. Will test with a few more modules before merging.
## Pre-merge
- [x] clearer comment with some background
- [x] safer prefix trimming

dgottlieb commented on module/modmanager/manager.go:
```diff
@@ -1170,6 +1171,16 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+// On windows only, this mutates socket paths so they work well with the GRPC library.
+// If you don't do this, it will confuse grpc-go's url.Parse call and surrounding logic.
+// See https://github.com/grpc/grpc-go/blob/v1.71.0/clientconn.go#L1720-L1727
+func cleanWindowsSocketPath(orig string) string {
```
It seems like this method expects inputs (from windows) such as:
```
C:\root\viam
```
And we want to mutate that into:
```
/root/viam
```
?

Can we add an example like that as a comment to help guide the reader?

abe-winter replied:
added

dgottlieb commented on module/modmanager/manager.go:
```diff
@@ -1067,7 +1068,7 @@ func (m *module) dial() error {
 	addrToDial := m.addr
 	if !rutils.TCPRegex.MatchString(addrToDial) {
-		addrToDial = "unix://" + m.addr
+		addrToDial = "unix://" + cleanWindowsSocketPath(addrToDial)
```
I see that the `m.addr` variable is "already cleaned" during [this assignment](https://github.com/viamrobotics/rdk/pull/4878/files#diff-a4308b77f384da81096eda21bf023471e80818dda86e74f9335f99a6a57d07c9R1202).

Did we observe more cleaning was needed here? Or is this just defensive?

abe-winter replied:
wasn't thinking about the full history of that member

good catch, removed

abe-winter commented:
note to reviewers -- acmorrow pointed out that this may not always be C drive, so I:
- added detection + error handling for non-C drive
- added a unit test

dgottlieb commented:
> note to reviewers -- acmorrow pointed out that this may not always be C drive

Way better than my feedback, thanks @acmorrow !
